jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AbstractMultiGridView.java

```diff
@@ -539,15 +539,16 @@ public ColumnSortList reloadColumnSortList() {
         if (gridPreferencesStore != null) {
             GridSortedColumnPreference gridSortedColumnPreference = gridPreferencesStore.getGridSortedColumnPreference();
             //Avoid duplicating the call push method of ColumnSortList when catch ColumnSortedEvent
-            //If repeating the call push method, there will throw 'IndexOutOfBoundsException: Row index: 0, Row size: 1'
             if (gridSortedColumnPreference != null && columnSortList.size() <= 1) {
                 Optional<ColumnMeta<T>> optional = getListGrid().getColumnMetaList().stream().filter(
                         tColumnMeta -> tColumnMeta.getColumn().getDataStoreName().equals(gridSortedColumnPreference.getDataStoreName()))
                         .findFirst();
                 if (optional.isPresent()) {
                     Column col = optional.get().getColumn();
                     col.setDataStoreName(gridSortedColumnPreference.getDataStoreName());
-                    columnSortList.push(new ColumnSortList.ColumnSortInfo(col, gridSortedColumnPreference.isAscending()));
+                    ColumnSortList sortedColumnPreferenceList = new ColumnSortList();
+                    sortedColumnPreferenceList.push(new ColumnSortList.ColumnSortInfo(col, gridSortedColumnPreference.isAscending()));
+                    return sortedColumnPreferenceList;
```

Comment:
I don't think you can simply return a new list. The view would still use the other list, see `getListGrid().getColumnSortList()`